core/src/test/scala/unit/kafka/log/LogCleanerIntegrationTest.scala

```diff
@@ -101,6 +101,8 @@ class LogCleanerIntegrationTest(compressionCodec: String) extends AbstractLogCle
       val messages = writeDups(numKeys = numKeys, numDups = 3, log = log, codec = codec)
       val startSize = log.size
 
+      log.onHighWatermarkIncremented(log.logEndOffset)
```

Comment:
This test case depends on segment deletion so it is necessary to advance the high watermark.